### ISSUE TITLE
Crash on start up

### ISSUE THREAD

**Issue by poofus** [BuildCraft/BuildCraft#2373]
Minecraft crashes on start up running BC 6.2 on MC 1.7.10

---- Minecraft Crash Report ----
// On the bright side, I bought you a teddy bear!

Time: 1/10/15 7:21 AM
Description: Initializing game

java.lang.VerifyError: Bad type on operand stack
Exception Details:
  Location:
    buildcraft/BuildCraftCore.load(Lcpw/mods/fml/common/event/FMLInitializationEvent;)V @184: invokespecial
  Reason:
    Type 'buildcraft/energy/BlockEngine' (current frame, stack[8]) is not assignable to 'net/minecraft/block/Block'
  Current Frame:
    bci: @184
    flags: { }
    locals: { 'buildcraft/BuildCraftCore', 'cpw/mods/fml/common/event/FMLInitializationEvent' }
    stack: { uninitialized 162, uninitialized 162, 'java/lang/String', 'java/lang/String', integer, integer, uninitialized 175, uninitialized 175, 'buildcraft/energy/BlockEngine', integer, integer }
  Bytecode:
    0x0000000: bb03 a259 1303 a413 03a5 0303 b201 a601
    0x0000010: b703 a8b6 03ac b303 aebb 03a2 5913 03b0
    0x0000020: 1303 b105 03b2 01b5 b203 aeb7 03a8 b603
    0x0000030: acb3 03b3 bb03 a259 1303 b513 03b6 0703
    0x0000040: b201 bab2 03b3 b703 a8b6 03ac b303 b8bb
    0x0000050: 03a2 5913 03ba 1303 bb10 0603 b201 bfb2
    0x0000060: 03b8 b703 a8b6 03ac b303 bdbb 03a2 5913
    0x0000070: 03bf 1303 c010 0803 b201 c4b2 03bd b703
    0x0000080: a8b6 03ac b303 c2bb 03a2 5913 03c4 1303
    0x0000090: c506 05b2 0160 b203 b3b7 03a8 b603 acb3
    0x00000a0: 03c7 bb03 a259 1303 c913 03ca 0410 febb
    0x00000b0: 01aa 59b2 03d0 0403 b703 d3b2 03ae b703
    0x00000c0: d6b6 03ac b303 d8bb 03a2 5913 03da 1303
    0x00000d0: db06 10fe bb01 aa59 b203 d004 04b7 03d3
    0x00000e0: b203 d8b7 03d6 b603 acb3 03dd bb03 a259
    0x00000f0: 1303 df13 03e0 0810 febb 01aa 59b2 03d0
    0x0000100: 0405 b703 d3b2 03dd b703 d6b6 03ac b303
    0x0000110: e2bb 03a2 5913 03e4 1303 e504 05b2 03eb
    0x0000120: b203 aeb7 03ee b603 acb3 03f0 bb03 a259
    0x0000130: 1303 f213 03f3 0805 b203 f7b2 03b8 b703
    0x0000140: eeb6 03ac b303 f9bb 03a2 5913 03fb 1303
    0x0000150: fc10 0905 b204 00b2 03c2 b703 eeb6 03ac
    0x0000160: b304 02bb 03a2 5913 0404 1304 0510 0705
    0x0000170: b204 0bb2 03bd b703 eeb6 03ac b304 0dbb
    0x0000180: 03a2 5913 040f 1304 1010 0910 feb2 0416
    0x0000190: b203 c2b7 03ee b603 acb3 0418 bb03 a259
    0x00001a0: 1304 1a13 041b 100a 03b2 041f b203 c2b7
    0x00001b0: 03ee b603 acb3 0421 bb03 a259 1304 2313
    0x00001c0: 0424 100b 10fe b204 28b2 0418 b703 eeb6
    0x00001d0: 03ac b304 2abb 03a2 5913 042c 1304 2d10
    0x00001e0: 0b05 b204 31b2 0402 b703 eeb6 03ac b304
    0x00001f0: 33bb 03a2 5913 0435 1304 3610 0d05 b204
    0x0000200: 3ab2 0433 b703 eeb6 03ac b304 3cbb 03a2
    0x0000210: 5913 043e 1304 3f10 0b07 b204 43b2 0433
    0x0000220: b703 a8b6 03ac b304 45bb 03a2 5913 0447
    0x0000230: 1304 4810 0d07 b204 4cb2 0445 b703 a8b6
    0x0000240: 03ac b304 4ebb 03a2 5913 0450 1304 5210
    0x0000250: 0f05 b204 56b2 043c b703 eeb6 03ac b304
    0x0000260: 58bb 045a 5913 045c 1015 bd03 a259 03b2
    0x0000270: 03ae 5359 04b2 03b3 5359 05b2 03b8 5359
    0x0000280: 06b2 03bd 5359 07b2 03c2 5359 08b2 03c7
    0x0000290: 5359 1006 b203 d853 5910 07b2 03dd 5359
    0x00002a0: 1008 b203 e253 5910 09b2 03f0 5359 100a
    0x00002b0: b203 f953 5910 0bb2 0402 5359 100c b204
    0x00002c0: 0d53 5910 0db2 0418 5359 100e b204 2153
    0x00002d0: 5910 0fb2 042a 5359 1010 b204 3353 5910
    0x00002e0: 11b2 043c 5359 1012 b204 4553 5910 13b2
    0x00002f0: 044e 5359 1014 b204 5853 b704 5fb3 0461
    0x0000300: b204 61b8 0465 b1                      

	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:344)
	at cpw.mods.fml.common.FMLModContainer.constructMod(FMLModContainer.java:420)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
	at cpw.mods.fml.common.LoadController.sendEventToModContainer(LoadController.java:208)
	at cpw.mods.fml.common.LoadController.propogateStateMessage(LoadController.java:187)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
	at cpw.mods.fml.common.LoadController.distributeStateMessage(LoadController.java:118)
	at cpw.mods.fml.common.Loader.loadMods(Loader.java:492)
	at cpw.mods.fml.client.FMLClientHandler.beginMinecraftLoading(FMLClientHandler.java:204)
	at net.minecraft.client.Minecraft.func_71384_a(Minecraft.java:480)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:867)
	at net.minecraft.client.main.Main.main(SourceFile:148)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Stacktrace:
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:344)
	at cpw.mods.fml.common.FMLModContainer.constructMod(FMLModContainer.java:420)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
	at cpw.mods.fml.common.LoadController.sendEventToModContainer(LoadController.java:208)
	at cpw.mods.fml.common.LoadController.propogateStateMessage(LoadController.java:187)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
	at cpw.mods.fml.common.LoadController.distributeStateMessage(LoadController.java:118)
	at cpw.mods.fml.common.Loader.loadMods(Loader.java:492)
	at cpw.mods.fml.client.FMLClientHandler.beginMinecraftLoading(FMLClientHandler.java:204)
	at net.minecraft.client.Minecraft.func_71384_a(Minecraft.java:480)

-- Initialization --
Details:
Stacktrace:
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:867)
	at net.minecraft.client.main.Main.main(SourceFile:148)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

-- System Details --
Details:
	Minecraft Version: 1.7.10
	Operating System: Windows 7 (amd64) version 6.1
	Java Version: 1.8.0_25, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 117632504 bytes (112 MB) / 266153984 bytes (253 MB) up to 1060372480 bytes (1011 MB)
	JVM Flags: 6 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx1G -XX:+UseConcMarkSweepGC -XX:+CMSIncrementalMode -XX:-UseAdaptiveSizePolicy -Xmn128M
	AABB Pool Size: 0 (0 bytes; 0 MB) allocated, 0 (0 bytes; 0 MB) used
	IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
	FML: MCP v9.05 FML v7.10.85.1230 Minecraft Forge 10.13.2.1230 9 mods loaded, 9 mods active
	mcp{9.05} [Minecraft Coder Pack] (minecraft.jar) Unloaded->Constructed
	FML{7.10.85.1230} [Forge Mod Loader] (forge-1.7.10-10.13.2.1230.jar) Unloaded->Constructed
	Forge{10.13.2.1230} [Minecraft Forge] (forge-1.7.10-10.13.2.1230.jar) Unloaded->Constructed
	BuildCraft|Core{6.2.6} [BuildCraft] (buildcraft-6.2.6.jar) Unloaded
	BuildCraft|Transport{6.2.6} [BC Transport] (buildcraft-6.2.6.jar) Unloaded
	BuildCraft|Silicon{6.2.6} [BC Silicon] (buildcraft-6.2.6.jar) Unloaded
	BuildCraft|Builders{6.2.6} [BC Builders] (buildcraft-6.2.6.jar) Unloaded
	BuildCraft|Energy{6.2.6} [BC Energy] (buildcraft-6.2.6.jar) Unloaded
	BuildCraft|Factory{6.2.6} [BC Factory] (buildcraft-6.2.6.jar) Unloaded
	Launched Version: 1.7.10-Forge10.13.2.1230
	LWJGL: 2.9.1
	OpenGL: AMD Radeon R9 200 Series GL version 4.3.12618 Compatibility Profile Context 13.251.0.0, ATI Technologies Inc.
	GL Caps: Using GL 1.3 multitexturing.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Anisotropic filtering is supported and maximum anisotropy is 16.
Shaders are available because OpenGL 2.1 is supported.

	Is Modded: Definitely; Client brand changed to 'fml,forge'
	Type: Client (map_client.txt)
	Resource Packs: []
	Current Language: English (US)
	Profiler Position: N/A (disabled)
	Vec3 Pool Size: 0 (0 bytes; 0 MB) allocated, 0 (0 bytes; 0 MB) used
	Anisotropic Filtering: Off (1)

**Comment by asiekierka** [comment]
What?

**Comment by Kubuxu** [comment]
I think someone has too much fun with ASM.

**Comment by hea3ven** [comment]
This is indeed a strange issue. I tried to reproduce it in both linux and in windows. On windows it worked fine. Though in linux, the game started fine, but with a black screen. Once I removed the "-XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump" jvm flag it started working again. I'd suggest trying different combinations of the jvm flags to see if it's one of them causing the issue.
